go_build_process_test.go

```diff
@@ -199,6 +199,66 @@ func testGoBuildProcess(t *testing.T, context spec.G, it spec.S) {
 		})
 	})
 
+	context("when workspaces should be used", func() {
+		it.Before(func() {
+			Expect(os.WriteFile(filepath.Join(workspacePath, "go.mod"), nil, 0644)).To(Succeed())
+			Expect(os.Mkdir(filepath.Join(workspacePath, "vendor"), os.ModePerm)).To(Succeed())
+		})
+
+		it("inits and uses the workspaces before executing the go build process", func() {
+			binaries, err := buildProcess.Execute(gobuild.GoBuildConfiguration{
+				Workspace:           workspacePath,
+				Output:              filepath.Join(layerPath, "bin"),
+				GoCache:             goCache,
+				Targets:             []string{"."},
+				WorkspaceUseModules: []string{"./some/module1", "./some/module2"},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(binaries).To(Equal([]string{
+				filepath.Join(layerPath, "bin", "some-dir"),
+			}))
+
+			Expect(filepath.Join(layerPath, "bin")).To(BeADirectory())
+
+			Expect(executions[0].Args).To(Equal([]string{
+				"work",
+				"init",
+			}))
+
+			Expect(executions[1].Args).To(Equal([]string{
+				"work",
+				"use",
+				"./some/module1",
+				"./some/module2",
+			}))
+
+			Expect(executions[2].Args).To(Equal([]string{
+				"build",
+				"-o", filepath.Join(layerPath, "bin"),
+				"-buildmode", "pie",
+				"-trimpath",
+				".",
+			}))
+
+			Expect(executions[3].Args).To(Equal([]string{
+				"list",
+				"--json",
+				".",
+			}))
+
+			Expect(executable.ExecuteCall.Receives.Execution.Dir).To(Equal(workspacePath))
+			Expect(executable.ExecuteCall.Receives.Execution.Env).To(ContainElement(fmt.Sprintf("GOCACHE=%s", goCache)))
+
+			Expect(logs).To(ContainLines(
+				"  Executing build process",
+				"    Running 'go work init'",
+				"    Running 'go work use ./some/module1 ./some/module2'",
+				fmt.Sprintf(`    Running 'go build -o %s -buildmode pie -trimpath .'`, filepath.Join(layerPath, "bin")),
+				"      Completed in 0s",
```

Comment:
🤔 It's consistently expecting `0s` while other tests want `1s`. I can't quite tell why.